.github/workflows/ci-py.yml

```diff
@@ -53,18 +52,13 @@ jobs:
         uses: mozilla-actions/sccache-action@v0.0.6
 
       - name: Set up uv
-        run: .github/script/install-uv.sh
-      - name: "Set up Python"
-        uses: actions/setup-python@v5
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Restore uv cache
-        uses: actions/cache@v4
-        with:
-          path: ${{ env.UV_CACHE_DIR }}
-          key: uv-${{ runner.os }}-${{ hashFiles('uv.lock') }}
-          restore-keys: |
-            uv-${{ runner.os }}
+          version: "0.4.18"
+          enable-cache: true
+      - name: Install Python
```

Comment:
do we know how fast this is compared to the default setup python action, any reason to prefer it?